### DISCUSSION TITLE
Multi account type arbitrage support

### DISCUSSION THREAD

**Discussion by sidnvy** [nautechsystems/nautilus_trader#423] [Q&A] [answered]
Hello NautilusTrader team 👋

Since `nautilus_trader.portfolio.portfolio.Portfolio` [currently limit one account per ExecutionClient](https://github.com/nautechsystems/nautilus_trader/blob/44068bc93c7f55a6e51553461e0f8fc8433d8187/nautilus_trader/portfolio/portfolio.pyx#L75) I just wanna make sure is that means we can't operate on binance spot account and binance future account at same time in same Strategy/Trader ?

Is there any plan to support this kind of multi account type strategy?

Thanks!

**Comment by cjdsellers** [comment, marked as answer]
Hi @sidnvy 

Thanks for your interest!

You've identified a current limitation of the platform. This was intentional to keep the complexity down during development, however the core system is becoming much more stable and this is again something we could consider getting up and running.

Right now there is an assumption that the `venue == client_id` and also each adapter is only connected to one account for the venue (this is what's causing the limitation).

This could potentially be solved by connecting to multiple sessions inside the clients, with some changes to the current single account per client impl. Or we could use multiple clients per venue, and somehow control the routing based on say the `account_id`.

Could you tell me a little more about your use case? Are you using sub-accounts?

**Reply by sidnvy** [reply]
Thanks for your answer.

The use case here is a basic spot future arbitrage strategy. I use one BINANCE account api key/sec pair which enabled both future trading and spot trading ability (not sub-account  only single normal account).

Since I can only set one CCXT-BINANCE execution client, the "defaultType" option will be either spot or future. I will try your suggestion above. 👍

**Reply by cjdsellers** [reply]
OK interesting. Given we're looking at the Binance integration right now, we'll keep this in mind and make sure its possible to trade on both through the one account.
